src/Symfony/Component/BrowserKit/Response.php

```diff
@@ -123,13 +123,14 @@ public function getHeaders()
      */
     public function getHeader($header, $first = true)
     {
+        $normalizedHeader = str_replace('-', '_', strtolower($header));
         foreach ($this->headers as $key => $value) {
-            if (str_replace('-', '_', strtolower($key)) == str_replace('-', '_', strtolower($header))) {
+            if (str_replace('-', '_', strtolower($key)) === $normalizedHeader) {
                 if ($first) {
                     return is_array($value) ? (count($value) ? $value[0] : '') : $value;
                 }
 
-                return is_array($value) ? $value : array($value);
+                return (array) $value;
```

Comment:
In fact, I'd remove both ternary and array-cast operators and replace them by an `if`. The reason being that both of them trigger memcopies. See https://www.drupal.org/node/1838368
So here and below:
```
if (!is_array($value)) {
    $value = array($value);
}

return $value;
```